test/unit/strategies/BaseStrategyUnit.t.sol

```diff
@@ -0,0 +1,328 @@
+
+        // It should call _afterWithdraw
+        baseStrategy.expectCall__afterWithdraw(_token, _amount, _recipient);
+
+        // It should emit event
+        vm.expectEmit();
+        emit IBaseStrategy.Withdrew(_token, _amount, _recipient);
+
+        // It should not change the pool amount
+        assertEq(baseStrategy.getPoolAmount(), poolAmountBefore);
+
+        baseStrategy.withdraw(_token, _amount, _recipient);
+    }
+
+    modifier whenTokenIsPoolToken(address _token, uint256 _amount, address _recipient, uint256 _contractBalance) {
+        vm.assume(_token != NATIVE);
+        vm.assume(_token != address(vm));
+        vm.assume(_token != 0x000000000000000000636F6e736F6c652e6c6f67);
+
+        baseStrategy.mock_call__checkOnlyPoolManager(address(this));
+
+        vm.mockCall(
+            _token,
+            abi.encodeWithSelector(IERC20.balanceOf.selector, address(baseStrategy)),
+            abi.encode(_contractBalance)
+        );
+        vm.mockCall(_token, abi.encodeWithSelector(IERC20.transfer.selector, _recipient, _amount), abi.encode(true));
+        vm.mockCall(
+            address(0),
+            abi.encodeWithSelector(IAllo.getPool.selector, 0),
+            abi.encode(
+                IAllo.Pool({
+                    profileId: bytes32(0),
+                    strategy: IBaseStrategy(address(0)),
+                    token: _token,
+                    metadata: Metadata({protocol: 0, pointer: ""}),
+                    managerRole: bytes32(0),
+                    adminRole: bytes32(0)
+                })
+            )
+        );
+        _;
+    }
+
+    function test_WithdrawWhenTokenIsPoolToken(
+        address _token,
+        uint256 _amount,
+        address _recipient,
+        uint256 _contractBalance,
+        uint256 _poolAmount
+    ) external whenTokenIsPoolToken(_token, _amount, _recipient, _contractBalance) {
+        vm.assume(_contractBalance > _amount);
+
+        _poolAmount = bound(_poolAmount, 0, _contractBalance - _amount);
+        baseStrategy.set__poolAmount(_poolAmount);
+
+        // It should call balanceOf at token
+        vm.expectCall(_token, abi.encodeWithSelector(IERC20.balanceOf.selector, address(baseStrategy)));
+
+        baseStrategy.withdraw(_token, _amount, _recipient);
+    }
+
+    function test_WithdrawRevertWhen_AmountIsGreaterThanPoolAmount(
+        address _token,
+        uint256 _amount,
+        address _recipient,
+        uint256 _contractBalance
+    ) external whenTokenIsPoolToken(_token, _amount, _recipient, _contractBalance) {
+        vm.assume(_amount > 0);
+        vm.assume(_contractBalance > _amount);
+
+        baseStrategy.set__poolAmount(_contractBalance);
+
+        // It should revert
+        vm.expectRevert(IBaseStrategy.BaseStrategy_WITHDRAW_MORE_THAN_POOL_AMOUNT.selector);
+
+        baseStrategy.withdraw(_token, _amount, _recipient);
+    }
```

Comment:
There should be three important variables in this test: poolAmount, balance and _amount. What do you think of something like this?
```suggestion
    function test_WithdrawRevertWhen_AmountIsGreaterThanPoolAmount(
        address _token,
        uint256 _amount,
        address _recipient,
        uint256 _contractBalance,
        uint256 _poolAmount
    ) external whenTokenIsPoolToken(_token, _amount, _recipient, _contractBalance) {
        _poolAmount = bound(_poolAmount, 1, _contractBalance);
        _amount = bound(_amount, _contractBalance - _poolAmount + 1, _contractBalance);

        baseStrategy.set__poolAmount(_poolAmount);

        // It should revert
        vm.expectRevert(IBaseStrategy.BaseStrategy_WITHDRAW_MORE_THAN_POOL_AMOUNT.selector);

        baseStrategy.withdraw(_token, _amount, _recipient);
    }
```